WooCommerce/Classes/Authentication/Store Creation/Profiler/StoreCreationProfilerQuestionContainerViewModel.swift

```diff
@@ -0,0 +1,113 @@
+        case .country:
+            return 0.6
+        case .challenges:
+            return 0.8
+        case .features:
+            return 1
+        }
+    }
+
+    var previousQuestion: StoreCreationProfilerQuestion? {
+        .init(rawValue: self.rawValue - 1)
+    }
+}
+
+/// View model for `StoreCreationProfilerQuestionContainer`.
+final class StoreCreationProfilerQuestionContainerViewModel: ObservableObject {
+
+    let storeName: String
+    private let analytics: Analytics
+    private let completionHandler: (SiteProfilerData?) -> Void
+
+    private var storeCategory: StoreCreationCategoryAnswer?
+    private var sellingStatus: StoreCreationSellingStatusAnswer?
+    private var storeCountry: SiteAddress.CountryCode = .US
+    private var challenges: [StoreCreationChallengesAnswer] = []
+    private var features: [StoreCreationFeaturesAnswer] = []
+
+    @Published private(set) var currentQuestion: StoreCreationProfilerQuestion = .sellingStatus
+
+    init(storeName: String,
+         analytics: Analytics = ServiceLocator.analytics,
+         onCompletion: @escaping (SiteProfilerData?) -> Void) {
+        self.storeName = storeName
+        self.analytics = analytics
+        self.completionHandler = onCompletion
+    }
+
+    func saveSellingStatus(_ answer: StoreCreationSellingStatusAnswer?) {
+        if answer == nil {
+            analytics.track(event: .StoreCreation.siteCreationProfilerQuestionSkipped(step: .profilerCategoryQuestion))
+        }
+        sellingStatus = answer
+        currentQuestion = .category
+    }
+
+    func saveCategory(_ answer: StoreCreationCategoryAnswer?) {
+        if answer == nil {
+            analytics.track(event: .StoreCreation.siteCreationProfilerQuestionSkipped(step: .profilerCategoryQuestion))
+        }
+        storeCategory = answer
+        currentQuestion = .country
+    }
+
+    func saveCountry(_ answer: SiteAddress.CountryCode) {
+        storeCountry = answer
+        currentQuestion = .challenges
+    }
+
+    func saveChallenges(_ answer: [StoreCreationChallengesAnswer]) {
+        if answer.isEmpty {
+            analytics.track(event: .StoreCreation.siteCreationProfilerQuestionSkipped(step: .profilerChallengesQuestion))
+        }
+        challenges = answer
+        currentQuestion = .features
+    }
+
+    func saveFeatures(_ answer: [StoreCreationFeaturesAnswer]) {
+        if answer.isEmpty {
+            analytics.track(event: .StoreCreation.siteCreationProfilerQuestionSkipped(step: .profilerFeaturesQuestion))
+        }
+        features = answer
+        handleCompletion()
+    }
+
+    func backtrackOrDismissProfiler() {
+        if let previousQuestion = currentQuestion.previousQuestion {
+            currentQuestion = previousQuestion
+        } else {
+            // TODO: show confirm alert if needed
+            completionHandler(nil)
+        }
+    }
+
+    private func handleCompletion() {
+        // TODO: update profiler data with challenges and features
```

Comment:
That makes sense! I didn't know whether we wanted to send the answers so added that comment just in case. Removed in 4abdd7f83a605fe8a3acbdd3eef3946c657a48e5.